### PR TITLE
Login and bookmark page.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* /bcgov/entity/10248

*Description of changes:*
1. Added Login page (route);
2. Handle bookmark function;
3. Handle route logic between Uber dashboard and ppr ui;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by doug-lovett — APPROVED** [review]
Looks good.

**Review by kialj876 — APPROVED** [review]
very nice 👍

File: ppr-ui/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "http-status-codes": "^2.1.4",
     "material-design-icons": "^3.0.1",
     "moment": "^2.29.1",
-    "sbc-common-components": "2.4.19",
+    "sbc-common-components": "2.4.33",
```
Comment (severinbeauvais) [inline ppr-ui/package.json]: This substantial change will require some additional (regression) testing around the common components used in this app.

Due to a lack of release notes, the tester should look at the commit history between v2.4.19 and v2.4.33 to identify potential breakages.

File: ppr-ui/public/config/configuration.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "REGISTRY_URL": "https://dev.bcregistry.ca/business/auth/",
+  "REGISTRY_URL": "https://yfthig-dev.web.app/dashboard",
```
Comment (severinbeauvais) [inline ppr-ui/public/config/configuration.json:2]: I assume the config variables in each environment will be updated...

File: ppr-ui/src/App.vue [hunk 3]
```diff
@@ -14,20 +14,11 @@
...
 
-    <!-- Initial Page Load Transition -->
-    <transition name="fade">
-      <div class="loading-container" v-show="!haveData">
-        <div class="loading__content">
-          <v-progress-circular color="primary" size="50" indeterminate />
-          <div class="loading-msg">Loading</div>
-        </div>
-      </div>
-    </transition>
```
Comment (severinbeauvais) [inline ppr-ui/src/App.vue:25]: If there is no more "spinner" in use anywhere in this app then its CSS (possibly in base) can be removed.

File: ppr-ui/src/main.ts [hunk 4]
```diff
@@ -58,9 +58,23 @@ async function start () {
     await initLdClient()
   }
 
-  // configure KeyCloak Service
-  console.info('Starting Keycloak service...') // eslint-disable-line no-console
-  await KeycloakService.setKeycloakConfigUrl(sessionStorage.getItem('KEYCLOAK_CONFIG_PATH'))
+  // Initialize Keycloak / sync SSO
+  await syncSession()
+
+  async function syncSession () {
```
Comment (severinbeauvais) [inline ppr-ui/src/main.ts]: This is currently a nested function, which is legal but usually makes the parent function pretty big.

I suggest either:
- move this function to the bottom of this file
- change this function to inline code

File: ppr-ui/src/router/router.ts [hunk 5]
```diff
@@ -56,5 +71,10 @@ export function getVueRouter () {
     return Boolean(route.name === 'signout')
   }
 
+  /** Returns True if route is Login success, else False. */
+  function isLoginSuccess (route: Route): boolean {
+    return Boolean(route.name === 'login' && route.hash)
```
Comment (severinbeauvais) [inline ppr-ui/src/router/router.ts]: Use your new `RouteNames` enum here.

File: ppr-ui/package.json (hunk 1 above)
Comment (pwei1018) [reply ppr-ui/package.json]: I will make a note to QA.

File: ppr-ui/public/config/configuration.json (hunk 2 above)
Comment (pwei1018) [reply ppr-ui/public/config/configuration.json:2]: Yes, it is in 1pass.

File: ppr-ui/package.json (hunk 1 above)
Comment (severinbeauvais) [reply ppr-ui/package.json]: Ok. (I do not have permissions to resolve this conversation.)

File: ppr-ui/public/config/configuration.json (hunk 2 above)
Comment (severinbeauvais) [reply ppr-ui/public/config/configuration.json:2]: Ok. (I do not have permissions to resolve this conversation.)

File: ppr-ui/src/main.ts (hunk 4 above)
Comment (severinbeauvais) [reply ppr-ui/src/main.ts]: Thanks for the fix.

Comment (severinbeauvais) [reply ppr-ui/src/main.ts]: (I do not have permissions to resolve this conversation.)

File: ppr-ui/src/router/router.ts (hunk 5 above)
Comment (severinbeauvais) [reply ppr-ui/src/router/router.ts]: Nice. Thanks.

(I do not have permissions to resolve this conversation.)

**Review by severinbeauvais — APPROVED** [review]
LGTM.

**Review by jordiwes — APPROVED** [review]

**Pull request merged**